src/librustc_trait_selection/traits/structural_match.rs

```diff
@@ -66,26 +67,25 @@ pub fn search_for_structural_match_violation<'tcx>(
 /// Note that this does *not* recursively check if the substructure of `adt_ty`
 /// implements the traits.
 pub fn type_marked_structural(
```

Comment:
The `type_marked_structural` method is now only called from the `has_structural_eq_impls` query closure, right?

I think it would be better to make it non-`pub` in that case.